src/Testcontainers.Azurite/AzuriteBuilder.cs

```diff
@@ -66,14 +68,30 @@ public override AzuriteContainer Build()
         return new AzuriteContainer(azuriteBuilder.DockerResourceConfiguration, TestcontainersSettings.Logger);
     }
 
+    public AzuriteBuilder WithAccountCredentials(string accountName, string accountKey)
+    {
+        return Merge(DockerResourceConfiguration, new AzuriteConfiguration(accountName: accountName, accountKey: accountKey))
+            .WithEnvironment("AZURITE_ACCOUNTS", $"{accountName}:{accountKey}");
+    }
+
+    public AzuriteBuilder WithPemCertificate(string certificatePath, string certificateKeyPath)
+    {
+        return Merge(DockerResourceConfiguration, new AzuriteConfiguration(certificatePath: certificatePath, certificateKeyPath: certificateKeyPath))
+            .WithBindMount(Path.GetDirectoryName(GetType().Assembly.Location), "/internal")
+            .WithCommand("--oauth", "basic")
+            .WithCommand("--cert", CleanUpPaths(certificatePath), "--key", CleanUpPaths(certificateKeyPath));
+    }
+
     /// <inheritdoc />
     protected override AzuriteBuilder Init()
     {
         return base.Init()
             .WithImage(AzuriteImage)
             .WithPortBinding(BlobPort, true)
             .WithPortBinding(QueuePort, true)
-            .WithPortBinding(TablePort, true);
+            .WithPortBinding(TablePort, true)
+            // Default command, if not specified the command will be completely overwritten in case of a certificate.
+            .WithCommand("azurite", "--blobHost", "0.0.0.0", "--queueHost", "0.0.0.0", "--tableHost", "0.0.0.0");
```

Comment:
We should consider the `_enabledServices` field here. This may require extending th `AzuriteService` class. We may not initiate every service in the future.